### PR TITLE
Boot: Sync user with redux state on change

### PR DESCRIPTION
**Pull request by aidvu**
We only set the user on boot. If it gets fetched/updated, `getCurrentUser` still returns the initial one.

Fixes: https://github.com/Automattic/wp-calypso/issues/11668

Test:
1. Create a new user
2. Create a domain-only site on https://wordpress.com/domains (pick a domain, than change URL to Calypso to hit the Flow)
3. Make sure there is no site switcher
4. Create another domain-only site
5. Make sure site switcher is visible w/o refresh

### CODE REVIEW

File: client/my-sites/current-site/index.jsx [hunk 1]
```diff
@@ -163,15 +167,17 @@ const CurrentSite = React.createClass( {
...
-		const selectedSiteId = getSelectedSiteId( state );
+		const selectedSiteId = getSelectedSiteId( state ),
+			user = getCurrentUser( state );
 
 		return {
+			isJetpack: isJetpackSite( state, selectedSiteId ),
 			selectedSiteId,
 			selectedSite: getSelectedSite( state ),
-			isJetpack: isJetpackSite( state, selectedSiteId )
+			siteCount: get( user, 'visible_site_count', 0 )
```
Comment (aduth) [inline client/my-sites/current-site/index.jsx]: Should this be a selector of its own?

**Comment by aidvu** [comment]
@aduth testing instructions added.

File: client/my-sites/current-site/index.jsx (hunk 1 above)
Comment (aidvu) [inline client/my-sites/current-site/index.jsx]: Hmm, would probably be a good idea! I'd need to refactor quite a few things tho. Maybe do it in a separate PR?
